### PR TITLE
[logs] Support Shift JIS encoded files

### PR DESCRIPTION
**Pull request by shigeya-dd**
### What does this PR do?

Adds Shift JIS (Japanese) encoding support when tailing from file.
It can be enabled by adding `encoding: shiftjis` in a log configuration.
About Shift JIS: [wikipedia](https://en.wikipedia.org/wiki/Shift_JIS)

### Motivation

Japanese versions of Windows uses Shift JIS as a primary encoding for Japanese.
Logs from apps and package softwares in Shift JIS cannot be handled by Datadog directly,
So currently, users need to forward logs to fluentd once, and translate to utf-8 with fluentd plugin, then forward them to Datadog.
With this patch, those users can send Shift JIS logs directly via Datadog agent.

### Additional Notes

Shift JIS is written as `ShiftJIS`, `Shift_JIS` and `Shift-JIS`. The underscore and hyphen might be annoying for users to configure correctly, so I chose no hyphenation `shiftjis` as an option string in the configuration.

### Describe how to test/QA your changes

Tail files encoded in Shift JIS. 
On linux system, generate some log lines with Shift JIS encoded message.
`echo "日本" | iconv -t shift_jis | logger`

```
logs:
        - type: file
          path: "/var/log/syslog"
          service: syslog-test
          source: syslog
          encoding: shift-jis
```
for Windows, I used `tomcat`. Set Java locale in `bin\catalina.bat` as the following
`set "JAVA_OPTS=%JAVA_OPTS% -Duser.language=ja -Duser.region=JP"`

and set log file encoding in the `conf\logging.properties`.
`1catalina.org.apache.juli.AsyncFileHandler.encoding = SJIS`

Then `logs\catalina*.log` will be written in Shift JIS. Encoding can be confirmed with editors.

image

With current released agent, Shift JIS logs are shown with garbled characters like the following

image

With this patch, logs show up as expected.

image

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by djmitche — CHANGES_REQUESTED** [review]
Thanks for this!

Shift-JIS has a general issue around searching for ascii characters, because they may appear as the second byte of a multibyte sequence.  The agent searches for newline characters (byte 0x0a), and a spurious newline detection would result splitting a log line in the middle of the line.  However, Shift-JIS does not use 0x0a as a second byte, so in this case the searching issue does not apply.

It took me some reading to figure this out, so I think it bears a code comment in `pkg/logs/input/file/tailer.go`.

Also note that the unit tests are requiring some very minor formatting fixes.

Other than that, this looks great and I'd be happy to merge it.

File: pkg/logs/config/integration_config.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ const (
 	UTF16BE string = "utf-16-be"
 	// UTF16LE for UTF-16 Little Endian encoding
 	UTF16LE string = "utf-16-le"
+	// SHIFTJIS for Shift JIS (Japanese) encoding
+	SHIFTJIS string = "shiftjis"
```
[the remark this answers is not in the document]
Comment (shigeya-dd) [reply pkg/logs/config/integration_config.go]: Updated to `shift-jis`. Thanks!

**Comment by shigeya-dd** [comment]
Hi, thank you for your review! I updated as followings,
- Replaced option string `shiftjis` with `shift-jis`
- Added some comment in `pkg/logs/input/file/tailer.go`
- Reformatted with `gofmt`

**Review by apigirl — COMMENTED** [review]
release note looks fine

**Comment by djmitche** [comment]
Awesome, thanks for the contribution!  I also added a docs link.  And I added shift-jis to my upcoming tech talk on encodings :)

**Review by djmitche — APPROVED** [review]

**Comment by shigeya-dd** [comment]
Thank you, @djmitche !!

**Pull request merged**